diff --git a/src/core/unix/SDL_gtk.c b/src/core/unix/SDL_gtk.c
index 32ae27d..777c85a 100644
--- a/src/core/unix/SDL_gtk.c
+++ b/src/core/unix/SDL_gtk.c
@@ -85,4 +85,4 @@
-#ifndef HAVE_GETRESUID
+#if !defined(HAVE_GETRESUID) && !defined(__linux__)
 // Non-POSIX, but Linux and some BSDs have it.
 // To reduce the number of code paths, if getresuid() isn't available at
 // compile-time, we behave as though it existed but failed at runtime.
@@ -93,4 +93,4 @@
-#ifndef HAVE_GETRESGID
+#if !defined(HAVE_GETRESGID) && !defined(__linux__)
 // Same as getresuid() but for the primary group
 static inline int getresgid(uid_t *ruid, uid_t *euid, uid_t *suid) {
     errno = ENOSYS;
